README.md

```diff
@@ -33,26 +33,56 @@ For [conda](https://docs.conda.io/) and [spack](https://spack.io/) installation
+First we'll import various classes and functions used in the following sections:
 
-We provide a script, `train.py` for training models using a subset of the datasets. We do this with the PyTorch Lightning `LightningModule`s and `LightningDataModule`s implemented under the `torchgeo.trainers` namespace.
-The `train.py` script is configurable via the command line and/or via YAML configuration files. See the [conf/](conf/) directory for example configuration files that can be customized for different training runs.
+```python
+from torch.utils.data import DataLoader
+from torchgeo.datasets import CDL, Landsat7, Landsat8, TropicalCycloneWindEstimation, stack_samples
+from torchgeo.samplers import RandomGeoSampler
+```
 
-```console
-$ python train.py config_file=conf/landcoverai.yaml
+### Composing geospatial datasets
+
+Many use cases involve working with geospatial data and combining it in intelligent ways. In this example, we assume that the user has Landsat 7 and 8 imagery downloaded. We first create a single dataset containing all Landsat imagery. We only use the Landsat 7 bands for Landsat 8 and take the union between both datasets.
+
+```python
+landsat7 = Landsat7(root="...")
+landsat8 = Landsat8(root="...", bands=landsat7.bands)
+landsat = landsat7 | landsat8
+```
+
+Next, we take the intersection between this dataset and the Cropland Data Layer (CDL) dataset. We want to take the intersection instead of the union to ensure that we only sample from regions that have both Landsat and CDL data. Note that we can automatically download and checksum CDL data. Also note that each of these datasets may contain files in different coordinate reference systems (CRS) or resolutions, but TorchGeo automatically ensures that a matching CRS and resolution is used.
+
+```python
+cdl = CDL(root="...", download=True, checksum=True)
+dataset = landsat & cdl
+```
+
+This dataset can now be used with a PyTorch data loader. In order to sample from this dataset using geospatial coordinates, we create a random sampler class.
+
+```python
+sampler = RandomGeoSampler(dataset, size=256, length=10000)
+dataloader = DataLoader(dataset, batch_size=128, sampler=sampler, collate_fn=stack_samples)
+```
+
+This data loader can now be used in your normal training/evaluation pipeline.
+
+```python
+for batch in dataloader:
+    # train a model, or make predictions using a pre-trained model
 ```
 
 ### Download and use the Tropical Cyclone Wind Estimation Competition dataset
 
 This dataset is from a competition hosted by [Driven Data](https://www.drivendata.org/) in collaboration with [Radiant Earth](https://www.radiant.earth/). See [here](https://www.drivendata.org/competitions/72/predict-wind-speeds/) for more information.
 
-Using this dataset in torchgeo is as simple as importing and instantiating the appropriate class.
+Using this dataset in TorchGeo is as simple as importing and instantiating the appropriate class.
```

Comment:
Do pytorch libraries have a convention for this?

Reply:
Nope. If you look at the [torchvision docs](https://pytorch.org/vision/stable/index.html), torchaudio, torchtext, and torchvision are all lowercase, whereas TorchElastic and TorchServe are camelcase. I don't really have a preference, but I would like to at least be consistent.